http-client/src/main/java/com/facebook/airlift/http/client/jetty/ChunkedBytesContentProvider.java

```diff
@@ -0,0 +1,95 @@
+package com.facebook.airlift.http.client.jetty;
+
+import org.eclipse.jetty.client.util.AbstractTypedContentProvider;
+
+import java.nio.ByteBuffer;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.Math.min;
+import static java.util.Objects.requireNonNull;
+
+public class ChunkedBytesContentProvider
+        extends AbstractTypedContentProvider
+{
+    private static final int DEFAULT_BUFFER_SIZE = 4096;
+
+    private final byte[] bytes;
+    private final int bufferSize;
+
+    public ChunkedBytesContentProvider(byte[] bytes)
+    {
+        this("application/octet-stream", bytes, DEFAULT_BUFFER_SIZE);
+    }
+
+    public ChunkedBytesContentProvider(byte[] bytes, int bufferSize)
+    {
+        this("application/octet-stream", bytes, bufferSize);
+    }
+
+    public ChunkedBytesContentProvider(String contentType, byte[] bytes)
+    {
+        this(contentType, bytes, DEFAULT_BUFFER_SIZE);
+    }
+
+    public ChunkedBytesContentProvider(String contentType, byte[] bytes, int bufferSize)
+    {
+        super(contentType);
+        this.bytes = requireNonNull(bytes, "bytes is null");
+        checkArgument(bufferSize > 0, "bufferSize must be greater then zero: %s", bufferSize);
+        this.bufferSize = bufferSize;
+    }
+
+    @Override
+    public long getLength()
+    {
+        return bytes.length;
+    }
+
+    @Override
+    public boolean isReproducible()
```

Comment:
curious: what does this mean?

Reply:
It is whether you can call ContentProvider#iterator() multiple times. I imagine there are some content providers that can generate content only once. For example is your content provider streams the data from the remote service, and you are authorized to read this content only once?